pkg/forwarder/forwarder.go

```diff
@@ -208,25 +207,47 @@ func NewOptions(keysPerDomain map[string][]string) *Options {
 		)
 		validationInterval = config.DefaultAPIKeyValidationInterval
 	}
+
 	const forwarderRetryQueueMaxSizeKey = "forwarder_retry_queue_max_size"
 	const forwarderRetryQueuePayloadsMaxSizeKey = "forwarder_retry_queue_payloads_max_size"
-	retryQueueSize := config.Datadog.GetInt(forwarderRetryQueueMaxSizeKey)
-	retryQueuePayloadsTotalMaxSize := config.Datadog.GetInt(forwarderRetryQueuePayloadsMaxSizeKey)
 
-	if retryQueueSize > 0 {
-		log.Warnf("'%s' is deprecated. It is recommended to use '%s' as it takes the payload sizes into account.", forwarderRetryQueueMaxSizeKey, forwarderRetryQueuePayloadsMaxSizeKey)
-		retryQueuePayloadsTotalMaxSize = 0
+	retryQueuePayloadsTotalMaxSize := 15 * 1024 * 1024
+	if config.Datadog.IsSet(forwarderRetryQueuePayloadsMaxSizeKey) {
+		retryQueuePayloadsTotalMaxSize = config.Datadog.GetInt(forwarderRetryQueuePayloadsMaxSizeKey)
 	}
 
-	return &Options{
+	option := &Options{
 		NumberOfWorkers:                config.Datadog.GetInt("forwarder_num_workers"),
-		RetryQueueSize:                 retryQueueSize,
-		RetryQueuePayloadsTotalMaxSize: retryQueuePayloadsTotalMaxSize,
 		DisableAPIKeyChecking:          false,
+		RetryQueuePayloadsTotalMaxSize: retryQueuePayloadsTotalMaxSize,
 		APIKeyValidationInterval:       time.Duration(validationInterval) * time.Minute,
 		KeysPerDomain:                  keysPerDomain,
 		ConnectionResetInterval:        time.Duration(config.Datadog.GetInt("forwarder_connection_reset_interval")) * time.Second,
 	}
+
+	if config.Datadog.IsSet(forwarderRetryQueueMaxSizeKey) {
+		if config.Datadog.IsSet(forwarderRetryQueuePayloadsMaxSizeKey) {
+			log.Warnf("'%v' is set, but as this setting is deprecated, '%v' is used instead.", forwarderRetryQueueMaxSizeKey, forwarderRetryQueuePayloadsMaxSizeKey)
+		} else {
```

Comment:
Do we need this ? If the old setting is set we could directly use it. That way we could leave the default value for `forwarder_retry_queue_payloads_max_size` in the configuration like every other default and it would simplify the logic here.

Reply:
To know whether `forwarder_retry_queue_payloads_max_size` is actually set (with `IsSet`), we need to _not_ set any default in the `config` package unfortunately. That's a limitation of the `config`/`viper` packages, we have a card in the backlog to improve this in the future.

Reply:
Sorry if my comment wasn't' clear.

We already removed the default for `forwarder_retry_queue_max_size` so we know when it's set or not. My comment was about the need to check if both `forwarder_retry_queue_max_size` and `forwarder_retry_queue_payloads_max_size` are set. We could simply print a warning if the old one, `forwarder_retry_queue_max_size`, is used and simplify the logic here and keep the default for `forwarder_retry_queue_payloads_max_size` in `pkg/config`.

something like this:

```go
const forwarderRetryQueueMaxSizeKey = "forwarder_retry_queue_max_size"
const forwarderRetryQueuePayloadsMaxSizeKey = "forwarder_retry_queue_payloads_max_size"

retryQueuePayloadsTotalMaxSize = config.Datadog.GetInt(forwarderRetryQueuePayloadsMaxSizeKey)

option := &Options{
	NumberOfWorkers:                config.Datadog.GetInt("forwarder_num_workers"),
	DisableAPIKeyChecking:          false,
	RetryQueuePayloadsTotalMaxSize: retryQueuePayloadsTotalMaxSize,
	APIKeyValidationInterval:       time.Duration(validationInterval) * time.Minute,
	KeysPerDomain:                  keysPerDomain,
	ConnectionResetInterval:        time.Duration(config.Datadog.GetInt("forwarder_connection_reset_interval")) * time.Second,
}

if config.Datadog.IsSet(forwarderRetryQueueMaxSizeKey) {
	forwarderRetryQueueMaxSize := config.Datadog.GetInt(forwarderRetryQueueMaxSizeKey)
	option.SetRetryQueuePayloadsTotalMaxSizeFromQueueMax(forwarderRetryQueueMaxSize)
	log.Warnf("'%v = %v' is used, but this setting is deprecated. '%v = %v' (%v * 2MB) is used instead as the maximum payload size is 2MB.",
		forwarderRetryQueueMaxSizeKey,
		forwarderRetryQueueMaxSize,
		forwarderRetryQueuePayloadsMaxSizeKey,
		option.RetryQueuePayloadsTotalMaxSize,
		forwarderRetryQueueMaxSize)
}
```

WDYT ? I think having the default in `pkg/config` is very useful for us and contributors.

Reply:
ok I understand your point, thanks for clarifying.

Your proposal effectively changes the order of precedence of the 2 options compared to what the PR does currently: instead of having the new option `forwarder_retry_queue_payloads_max_size` take precedence over the old option `forwarder_retry_queue_max_size`, your proposed change would invert this order of precedence which is not ideal IMO (I would prefer for the new option to take precedence over the old one).

Not having the default of `forwarder_retry_queue_payloads_max_size` defined in `pkg/config` is an inconvenience I agree, but the default will still be documented in the `config_template.yaml` file. I think we should add comments in `pkg/config` to make that clearer in the code as well.

Ultimately, I think we should aim for the behavior that makes the most sense here, and solve the inconvenience of not defining all the defaults `pkg/config` later when we work on the improvements to the `pkg/config` package that we have in the backlog (to allow making a difference between options that are actually set, and options that only have a default value). WDYT?

Reply:
I think customers setting both settings will not be a common thing. But I'm OK with adding it to the TODO list.